### PR TITLE
ContainerChanges is not implemented by snapshotter-based ImageService

### PR DESCRIPTION
**Pull request by ndeloof**

### CODE REVIEW

File: daemon/changes.go [hunk 1]
```diff
@@ -21,6 +21,9 @@ func (daemon *Daemon) ContainerChanges(name string) ([]archive.Change, error) {
 
 	container.Lock()
 	defer container.Unlock()
+	if daemon.UsesSnapshotter() {
+		panic("not implemented")
+	}
 	if container.RWLayer == nil {
 		return nil, errors.New("RWLayer of container " + name + " is unexpectedly nil")
```
Comment (vvoland) [inline daemon/changes.go:28]: Why do we need a panic? Won't it just end with error here?

Comment (ndeloof) [reply daemon/changes.go:28]: for consistency with the other places where ImageService func are not implemented

Comment (tonistiigi) [reply daemon/changes.go:28]: I agree that if this is triggered by `docker diff` on the demo release then it shouldn't panic the daemon but return "not implemented" error to the user.

Comment (ndeloof) [reply daemon/changes.go:28]: then all other places the snapshotter-based ImageService does not implement a func should do the same

**Review by vvoland — APPROVED** [review]

**Comment by ndeloof** [comment]
upstream PR https://github.com/moby/moby/pull/44068

**Pull request merged**